lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -205,6 +204,18 @@ newTransactionLayer _proxy _protocolMagic epochLength = TransactionLayer
 
         pure $ toSealed $ SL.Tx unsigned wits scriptWits metadata
 
+    _decodeSignedTx
+        :: ByteString
+        -> Either ErrDecodeSignedTx (Tx, SealedTx)
+    _decodeSignedTx bytes = do
+        case Cardano.txSignedFromCBOR bytes of
+            Right (Cardano.TxSignedShelley txValid) -> pure $ toSealed txValid
+            Right (Cardano.TxSignedByron{}) ->
+                Left $ ErrDecodeSignedTxWrongPayload
+                "Detected Byron signed tx. Was expecting Shelley signed one!"
```

Comment:
I think that `TxSignedByron` refers to Shelley transactions using Byron addresses (and requiring special Byron UTxO witnesses). They'd therefore be so-to-speak, valid in Shelley.

Reply:
the full constructor looks like that : 
```
TxSignedByron !ByronTxBody !ByteString !(Byron.Hash ByronTxBody) !(Vector Byron.TxInWitness)
```
which supports what you wrote. Probably will leave it as TO_DO at first step

Reply:
Not _yet_ because those witnesses aren't yet supported in the node (although they landed in the ledger recently). So we will. Soon.